### PR TITLE
Fix comments being duplicated on refresh

### PR DESCRIPTION
**Pull request by cardoso**
This bug can be easily replicated by running the sample and refreshing an activity with comments

### CODE REVIEW

**Comment by cardoso** [comment]
The bug happens because it's not reaching this path in ReactionPaginator's load function due to the Pagination being .limit(100):
```swift
if case .none = pagination {
    self.items = []
    self.next = .none
}
```

So I just changed it to .none (default) like it's done for the FlatFeed .

**Comment by b-onc** [comment]
@cardoso Your observation is right, paginator is not removing items when `load` is called with a pagination option. IMHO this is a bug, `load` should not `append` items, it should load from start and `loadNext` should append the items.
So
```swift
if case .none = pagination {
  self.items = []
  self.next = .none
}
```
should be
```swift
if pagination != next {
  self.items = []
  self.next = .none
}
``` 
or a similar logic to make sure `load` and `loadNext` behaves differently.
Since this changes a lot of things (Pagination is not Equatable), I suggest introducing `reset` function to `PaginatorProtocol` to reset items (`items = []`) and putting these calls before `load(.limit(100))`s in `DetailViewController`, which will make sure we won't get replicated items.
 This bug is probably affecting other places but we don't know yet, this is the easiest and safest solution.

**Comment by cardoso** [comment]
@b-onc changes made

**Review by b-onc — CHANGES_REQUESTED** [review]
You have to add a `reset` call to `DetailViewController.swift:L377` and `L:492` too

**Comment by cardoso** [comment]
@b-onc nice catch. Since it's in so many places, I also introduced separate function for it.

File: Sources/Components/Flat Feed/DetailViewController.swift [hunk 1]
```diff
@@ -140,6 +140,11 @@ open class DetailViewController<T: ActivityProtocol>: BaseFlatFeedViewController
         self.sectionsData = sectionsData
     }
     
+    private func resetComments() {
```
Comment (b-onc) [inline Sources/Components/Flat Feed/DetailViewController.swift]: I suggest renaming this to `reloadComment`

**Review by b-onc — CHANGES_REQUESTED** [review]
The commit history is polluted, it has commits from PR #7. Can you rebase so it has only has commits related to this PR?

**Comment by cardoso** [comment]
Should be fine now @b-onc

**Review by b-onc — APPROVED** [review]

**Pull request merged**